src/dsp_tools/commands/xmlupload/stash/upload_stashed_resptr_props.py

```diff
@@ -31,25 +33,74 @@ def upload_stashed_resptr_props(
     logger.info("Upload the stashed resptrs...")
+
     upload_state.pending_stash = cast(Stash, upload_state.pending_stash)
     link_value_stash = cast(LinkValueStash, upload_state.pending_stash.link_value_stash)
-    for res_id, stash_items in link_value_stash.res_2_stash_items.copy().items():
-        res_iri = upload_state.iri_resolver.get(res_id)
+
+    _upload_process(link_value_stash=link_value_stash, iri_resolver=upload_state.iri_resolver, con=con, context=context)
+    _remove_empty_ids(link_value_stash=link_value_stash)
+
+
+def _upload_process(
+    link_value_stash: LinkValueStash,
+    iri_resolver: IriResolver,
+    con: Connection,
+    context: dict[str, str],
+) -> None:
+    """
+    Try to upload the stashed items. If the target_iri of the item can be found
+    and the item can then successfully be uploaded it is removed from link_value_stash.
+    The link_value_stash is updated accordingly, as a side effect.
+
+    Args:
+        link_value_stash: the current state of the link value stash
+        iri_resolver: service for resolving internal IDs to IRIs
+        con: connection to DSP
+        context: the JSON-LD context of the resource
+    """
+    for res_id, stash_items in link_value_stash.res_2_stash_items.items():
+        res_iri = iri_resolver.get(res_id)
         if not res_iri:
             # resource could not be uploaded to DSP, so the stash cannot be uploaded either
             # no action necessary: this resource will remain in nonapplied_resptr_props,
             # which will be handled by the caller
             continue
         print(f"{datetime.now()}:   Upload resptrs of resource '{res_id}'...")
         logger.info(f"  Upload resptrs of resource '{res_id}'...")
-        for stash_item in stash_items:
-            target_iri = upload_state.iri_resolver.get(stash_item.target_id)
+
+        start_position = len(stash_items) - 1
+        while start_position >= 0:
+            # taking values by index from the list of 'stash_items' should avoid the problem we encounter
+            # when looping over the same list we simultaneously remove from
+            # counting max-length to 0 avoids problem with diminishing list-length
+            stash_item = stash_items[start_position]
+
+            start_position -= 1
+
+            target_iri = iri_resolver.get(stash_item.target_id)
             if not target_iri:
                 continue
-            if _upload_stash_item(stash_item, res_iri, target_iri, con, context):
-                link_value_stash.res_2_stash_items[res_id].remove(stash_item)
-        if not link_value_stash.res_2_stash_items[res_id]:
-            del link_value_stash.res_2_stash_items[res_id]
+            if not _upload_stash_item(
+                stash=stash_item, res_iri=res_iri, target_iri=target_iri, con=con, context=context
+            ):
+                continue
+            link_value_stash.res_2_stash_items[res_id].remove(stash_item)
```

Comment:
Could be simplified to:

```suggestion
            if _upload_stash_item(
                stash=stash_item, res_iri=res_iri, target_iri=target_iri, con=con, context=context
            ):
                link_value_stash.res_2_stash_items[res_id].remove(stash_item)
```